sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/jsonschema/JsonSchemaResourceIntegrationTest.java

```diff
@@ -0,0 +1,267 @@
+ */
+
+package io.stargate.sgv2.docsapi.api.v2.namespaces.collections.jsonschema;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import io.stargate.sgv2.common.cql.builder.Replication;
+import io.stargate.sgv2.docsapi.config.constants.Constants;
+import io.stargate.sgv2.docsapi.service.schema.NamespaceManager;
+import io.stargate.sgv2.docsapi.service.schema.TableManager;
+import io.stargate.sgv2.docsapi.testprofiles.IntegrationTestProfile;
+import java.time.Duration;
+import javax.enterprise.context.control.ActivateRequestContext;
+import javax.inject.Inject;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.ClassOrderer;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestClassOrder;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@QuarkusTest
+@TestProfile(IntegrationTestProfile.class)
+@ActivateRequestContext
+@TestClassOrder(ClassOrderer.OrderAnnotation.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class JsonSchemaResourceIntegrationTest {
+
+  // TODO replace with HTTP call and remove @Inject to support end-to-end
+  //  remove @ActivateRequestContext
+  //  remove @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  //  drop keyspace properly, see https://github.com/quarkusio/quarkus/issues/25812
+  //  could be as the junit extension that runs before all tests
+
+  // base path for the test
+  public static final String BASE_PATH =
+      "/v2/namespaces/{namespace}/collections/{collection}/json-schema";
+  public static final String DEFAULT_NAMESPACE = RandomStringUtils.randomAlphanumeric(16);
+  public static final String DEFAULT_COLLECTION = RandomStringUtils.randomAlphanumeric(16);
+
+  @Inject NamespaceManager namespaceManager;
+  @Inject TableManager tableManager;
+  @Inject ObjectMapper objectMapper;
+
+  @BeforeAll
+  public void init() {
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+
+    namespaceManager
+        .createNamespace(DEFAULT_NAMESPACE, Replication.simpleStrategy(1))
+        .await()
+        .atMost(Duration.ofSeconds(10));
+
+    tableManager
+        .createCollectionTable(DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+        .await()
+        .atMost(Duration.ofSeconds(10));
+  }
+
+  @Nested
+  @Order(1)
+  @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+  class PutJsonSchema {
+
+    @Test
+    @Order(1)
+    public void happyPath() throws JsonProcessingException {
```

Comment:
updating of the schema is not tested, we should add just for the case of deeper int tests against different data stores.. or maybe we don't need it..

Reply:
I added a test that replaces one schema with another